test/suite/test_log04.py

```diff
@@ -0,0 +1,159 @@
+
+        # Set oldest and stable timestamps to 10.
+        self.conn.set_timestamp(
+            'oldest_timestamp=' + self.timestamp_str(10) +
+            ',stable_timestamp=' + self.timestamp_str(10))
+
+        key = ds_ts.key(10)
+        value10 = ds_ts.value(10)
+
+        # Confirm initial data at timestamp 10.
+        self.check(c_log, 10, key, value10)
+        self.check(c_ts, 10, key, value10)
+        self.check(c_nots, 10, key, value10)
+
+        # Update and then rollback.
+        value50 = ds_ts.value(50)
+        self.session.begin_transaction()
+        c_log[key] = value50
+        c_ts[key] = value50
+        c_nots[key] = value50
+        self.session.rollback_transaction()
+
+        # Confirm data at time 10 and 20.
+        self.check(c_log, 10, key, value10)
+        self.check(c_ts, 10, key, value10)
+        self.check(c_nots, 10, key, value10)
+        self.check(c_log, 20, key, value10)
+        self.check(c_ts, 20, key, value10)
+        self.check(c_nots, 20, key, value10)
+
+        # Update and then commit data at time 20.
+        value55 = ds_ts.value(55)
+        self.session.begin_transaction()
+        c_log[key] = value55
+        c_ts[key] = value55
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(20))
+        self.session.begin_transaction()
+        c_nots[key] = value55
+        self.session.commit_transaction()
+
+        # Confirm data at time 10 and 20.
+        self.check(c_log, 10, key, value55)
+        self.check(c_ts, 10, key, value10)
+        self.check(c_nots, 10, key, value55)
+        self.check(c_log, 20, key, value55)
+        self.check(c_nots, 20, key, value55)
+
+        # Update and then commit data at time 30.
+        value60 = ds_ts.value(60)
+        self.session.begin_transaction()
+        c_log[key] = value60
+        c_ts[key] = value60
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(30))
+        self.session.begin_transaction()
+        c_nots[key] = value60
+        self.session.commit_transaction()
+
+        # Confirm data at time 20 and 30
+        self.check(c_log, 20, key, value60)
+        self.check(c_ts, 20, key, value55)
+        self.check(c_nots, 20, key, value60)
+        self.check(c_log, 30, key, value60)
+        self.check(c_ts, 30, key, value60)
+        self.check(c_nots, 30, key, value60)
+
+        # Move the stable timestamp to 25. Checkpoint and rollback to a timestamp.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(25))
+        if self.ckpt:
+            self.session.checkpoint()
+        self.conn.rollback_to_stable()
+
+        # Confirm data at time 20 and 30.
+        self.check(c_log, 20, key, value60)
+        self.check(c_ts, 20, key, value55)
+        self.check(c_nots, 20, key, value60)
+        self.check(c_log, 30, key, value60)
+        self.check(c_ts, 30, key, value55)
+        self.check(c_nots, 30, key, value60)
```

Comment:
It would be testing that a restart is properly retaining the timestamp, not timestamp and recovery/replay is applying all the necessary information. (Note I did not say you should checkpoint there, just flush the log.)